metadata/src/main/java/org/apache/kafka/timeline/SnapshotRegistry.java

```diff
@@ -178,14 +178,18 @@ public Snapshot getSnapshot(long epoch) {
     /**
      * Creates a new snapshot at the given epoch.
      *
+     * If {@code epoch} already exists and it is the last snapshot then just return that snapshot.
```

Comment:
With this modification, the function should be renamed something like `getOrCreateSnapshot`, right?